src/TensorFlowNET.Core/Status/Status.cs

```diff
@@ -25,26 +26,37 @@ namespace Tensorflow
     /// TF_Status holds error information. It either has an OK code, or
     /// else an error code with an associated error message.
     /// </summary>
-    public class Status : DisposableObject
+    public sealed class Status : IDisposable
     {
         /// <summary>
         /// Error message
         /// </summary>
-        public string Message => c_api.StringPiece(TF_Message(_handle));
+        public string Message
+        {
+            get
+            {
+                using (Handle.Lease())
+                {
+                    return StringPiece(TF_Message(Handle));
+                }
```

Comment:
The .NET runtime will automatically acquire a lease when the `SafeHandle` is passed to `TF_Message`. However, the return value from `TF_Message` is an `IntPtr` which is also tied to the lifetime of the `SafeStatusHandle`. By wrapping the use of this pointer in a lease, we ensure the `SafeStatusHandle` is not finalized prior to the completion of the call to `StringPiece`.

📝 It's not normally desirable to use `Handle.Lease()` in coding patterns, but for cases where we _can't_ automatically have things work through proper P/Invoke signatures, it allows the code to still be reliable.